src/api/tokenList/TokenListApiImpl.ts

```diff
@@ -15,29 +11,47 @@ function _getImageUrl(tokenAddress?: string): string | undefined {
 export class TokenListApiImpl implements TokenList {
   public networkIds: number[]
   private _tokensByNetwork: { [networkId: number]: TokenDetails[] }
+  private _exchangeApi: ExchangeApi
 
-  public constructor(networkIds: number[]) {
+  public constructor(networkIds: number[], exchangeApi: ExchangeApi) {
```

Comment:
This is a bit tricky, I would think it's not bad for now, but the issue is "Api" is a logical layer we created to abstract certain things from the UI logic since the UI logic is complicated enough. 

Ideally, a layer relies **only in the immediate layer below** which for simplicity we don't have in this project for now. Only utils and libs.

In this case we have an Api depending on another Api. This would likely bring problems in the future.

Another approach would be:
- We abstract the access to blockchain, the file list, or whatever undelying persistence layer with a `Repository` layer. 
- We abstract our logics, denormalizing, joining data from different datasources (lists, TCRs, different smart contracts, etc..) into more UI like objects, etc... This would be a service layer (we can call it `Service` or `Api`). This layer would receive Repositories and config params in it's constructor, never other elements from the same layer.

Issue: I don't want to overcomplicate, so I'm happy to leave it like this if we don't have the issue yet. Also, another issue, would be that in may cases, our service layer might be just re-exposing some/all of the repository methods.

Happy to hear everyone's thoughts on alternatives.

Reply:
I'm not entirely sure what would be the immediate solution.

We already have a dependency between APIs, although they are a bit more related (Deposit and Exchange APIs require ERC20 api instance).

And if we want to update the token list with tokens on the exchange and not on our initial list, we'll need: tokenList (API), erc20 API and exchange API.

I agree and like the separation of concerns between the APIs, although I don't like the duplication it can create.